test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/FormActionTagHelperTest.cs

```diff
@@ -0,0 +1,387 @@
+                .Returns("admin/dashboard/index")
+                .Callback<UrlActionContext>(param => Assert.Equal(param.Values, expectedRouteValues))
+                .Verifiable();
+
+            var urlHelperFactory = new Mock<IUrlHelperFactory>();
+            urlHelperFactory
+                .Setup(f => f.GetUrlHelper(It.IsAny<ActionContext>()))
+                .Returns(urlHelper.Object);
+
+            var tagHelper = new FormActionTagHelper(urlHelperFactory.Object)
+            {
+                Action = "Index",
+                Controller = "Dashboard",
+                Area = "Admin",
+            };
+
+            // Act
+            await tagHelper.ProcessAsync(context, output);
+
+            // Assert
+            urlHelper.Verify();
+            Assert.Equal("submit", output.TagName);
+            Assert.Equal(1, output.Attributes.Count);
+            var attribute = Assert.Single(output.Attributes, attr => attr.Name.Equals("formaction"));
+            Assert.Equal("admin/dashboard/index", attribute.Value);
+            Assert.True(output.Content.GetContent().Length == 0);
+        }
+
+        [Fact]
+        public async Task ProcessAsync_AspAreaOverridesAspRouteArea()
+        {
+            // Arrange
+            var context = new TagHelperContext(
+                allAttributes: new TagHelperAttributeList(
+                    Enumerable.Empty<TagHelperAttribute>()),
+                items: new Dictionary<object, object>(),
+                uniqueId: "test");
+            var output = new TagHelperOutput(
+                "button",
+                attributes: new TagHelperAttributeList(),
+                getChildContentAsync: (useCachedResult, encoder) =>
+                {
+                    var tagHelperContent = new DefaultTagHelperContent();
+                    tagHelperContent.SetContent("Something");
+                    return Task.FromResult<TagHelperContent>(tagHelperContent);
+                });
+            output.Content.SetContent(string.Empty);
+
+            var expectedRouteValues = new RouteValueDictionary(new Dictionary<string, string> { { "area", "Admin" } });
+            var urlHelper = new Mock<IUrlHelper>();
+            urlHelper
+                .Setup(mock => mock.Action(It.IsAny<UrlActionContext>()))
+                .Returns("admin/dashboard/index")
+                .Callback<UrlActionContext>(param => Assert.Equal(param.Values, expectedRouteValues))
+                .Verifiable();
+
+            var urlHelperFactory = new Mock<IUrlHelperFactory>();
+            urlHelperFactory
+                .Setup(f => f.GetUrlHelper(It.IsAny<ActionContext>()))
+                .Returns(urlHelper.Object);
+
+            var tagHelper = new FormActionTagHelper(urlHelperFactory.Object)
+            {
+                Action = "Index",
+                Controller = "Dashboard",
+                Area = "Admin",
+                RouteValues = new Dictionary<string, string> { { "area", "Home" } }
```

Comment:
Couldn't this test be pulled up into the above one by adding this single line? I don't feel that that the conceptual pieces would be that drastically difference to cause a separate test red flag.